api/v1alpha1/dnsrecord_types.go

```diff
@@ -103,7 +103,8 @@ type DNSRecordSpec struct {
 	// +kubebuilder:validation:Pattern=`^(?:[\w\-.~:\/?#[\]@!$&'()*+,;=]+)\.(?:[\w\-.~:\/?#[\]@!$&'()*+,;=]+)$`
 	RootHost string `json:"rootHost"`
 
-	// providerRef is a reference to a provider secret.
+	// ProviderRef is a reference to a provider secret.
+	// +optional
 	ProviderRef ProviderRef `json:"providerRef"`
```

Comment:
Should this be a pointer if its now optional?

Reply:
Made the `Spec` one a pointer (the `Status` reference is not a pointer) following API conventions